### ISSUE TITLE
feat: ZeroTier Central migration

### ISSUE THREAD

**Issue by jakoberpf** [dec0dOS/zero-ui#24]
# Feature Request

**Describe the Feature Request**
When migrating from the hosted zerotier controller, it would be nice be able to add the already existing networks to the selfhosted controller. 

**Describe Preferred Solution**
To my knowledge of how zerotier works, i thought this would be possible by just joining the selfhosted controller to the zerotier network. This could be then done by either the cli in the pod or the UI. But so far i didn't manage to get the UI to show the network if the controller is joined. Or maybe my thinking process is flaud.

Does anyone have any experience with migrating from the hosted zerotier controller?

**If the feature request is approved, would you be willing to submit a PR?**
Yes

**Comment by JamesTX10** [comment]
> The full migration could be possible without rejoining the network on the client-side only from ztncui or any other self-hosted solution.
I am using ztncui currently. Is there a way to migrate those networks over to Zero-ui?

**Comment by jakoberpf** [comment]
@JamesTX10 @dec0dOS Sorry, didn't really got to his yet. I am still on the my.zerotier controller and still planning to work on this migration functionality...

**Comment by dec0dOS** [comment]
Migrating to the self-hosted solution is really painful, as it requires client-side reconfiguration.
Even as a zero-ui maintainer, I am currently using the ZeroTier Central for some networks.

**Comment by Zeitkind** [comment]
Is there a way to migrate from my ztncui?
I don't need all those fancy enterprise features, so I'd like to migrate my last bunch of networks and clients to a private server. I have a ztncui running for years, but development seemed to have stopped. I still have about 40 clients on zerotier.com and about another 40 on my ztncui. Every time they changed their license and reduced the free members, I created new networks on my ztncui and re-added all manually. Those 40 remaining are a pita to renew, so I'd really like to migrate them without much problems. Any ideas?

**Comment by dec0dOS** [comment]
@Zeitkind check the following discussion about the ztncui migration. 
https://github.com/dec0dOS/zero-ui/discussions/3

You could use your old zerotier controller key, so you don't have to manually reconfigure the clients.

**Comment by Zeitkind** [comment]
Will read it. Not sure yet, I don't really like nor use docker, I prefer a plain old repository or snap, but prob. try it on a VM host, hmmm..
Any ideas how to migrate from zerotier.com? Almost all clients are private and non commercial, so i don't really need any of those new fancy features, but there are a lot - and a lot of people that have no idea how to set up a new connection..  ^^ My goal is to completely go away from zerotier.com. I like their software, but tbh not their politics and total lack of communication.

**Comment by dec0dOS** [comment]
@Zeitkind,
No, it is not possible to migrate from ZeroTier Central without rejoin on client side as ZeroTier Central controller private keys are unknown
https://github.com/dec0dOS/zero-ui/issues/24#issuecomment-944274854

**Comment by Zeitkind** [comment]
Hmm.. ok, then I have to do it the hard way. sigh.
